src/app/shared/state/entry-wizard.service.ts

```diff
@@ -0,0 +1,141 @@
+  constructor(private entryWizardStore: EntryWizardStore, private defaultService: DefaultService, private alertService: AlertService) {}
+
+  /**
+   * Updates the list of git registries
+   */
+  updateGitRegistries() {
+    this.entryWizardStore.setLoading(true);
+    this.defaultService
+      .getUserRegistries()
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (registries: Array<string>) => {
+          this.entryWizardStore.update({ gitRegistries: registries });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Updates the list of organizations based on the given git registry
+   * @param registry git registry
+   */
+  updateGitOrganizations(registry: string) {
+    this.entryWizardStore.setLoading(true);
+    this.entryWizardStore.update({ gitOrganizations: undefined, gitRepositories: undefined });
+    const registryEnum = this.convertSourceControlStringToEnum(registry);
+    this.defaultService
+      .getUserOrganizations(registryEnum)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (organizations: Array<string>) => {
+          this.entryWizardStore.update({ gitOrganizations: organizations });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Updates the list of repositories based on the given git registry and organization
+   * @param registry git registry
+   * @param organization git organization
+   */
+  updateGitRepositories(registry: string, organization: string) {
+    this.entryWizardStore.setLoading(true);
+    const registryEnum = this.convertSourceControlStringToEnum(registry);
+
+    this.defaultService
+      .getUserOrganizationRepositories(registryEnum, organization)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (repositories: Array<Repository>) => {
+          this.entryWizardStore.update({ gitRepositories: repositories });
+        },
+        (error: HttpErrorResponse) => {
+          this.alertService.detailedError(error);
+        }
+      );
+  }
+
+  /**
+   * Adds a workflow for the given git repository
+   * @param registry git registry
+   * @param organization git organization
+   * @param repositoryName git repository name
+   */
+  addWorkflow(registry: string, organization: string, repositoryName: string) {
+    this.entryWizardStore.setLoading(true);
+    const registryEnum = this.convertSourceControlStringToEnum(registry);
+    this.defaultService
+      .addWorkflow(registryEnum, organization, repositoryName)
+      .pipe(finalize(() => this.entryWizardStore.setLoading(false)))
+      .subscribe(
+        (workflow: BioWorkflow) => {
+          this.alertService.detailedSuccess('Workflow ' + registry + '/' + organization + '/' + repositoryName + ' has been added');
```

Comment:
Does the store need to be updated?

Reply:
By store do you mean the list of workflows on the page (my-workflows)? This is updated when you click finish. I'm open to more changes in the next round of updates once people get the chance to try it out.

Reply:
I meant the Akita store:

1. You get list of repos from web service and put them into entity-wizard.store
2. The UI displays the (filtered) repos
3. User toggles a repo
4. This code is called to toggle repo on backend
5. Doesn't store on step 1 need to have new state of repo?

Reply:
Good point, I've made it pull changes on error so that the toggles are correct.